File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        gasoline=nums[0]
        if len(nums)==1 :
            return True
        for i in range(1,len(nums)-1):
            gasoline-=1
            if gasoline==-1:
                return False
            if gasoline<nums[i]:
                gasoline=nums[i]
        if gasoline>=1:
            return True
        else:
            return False
        
        

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        if(nums.length==1){
            return true;
        }
        int gasoline=nums[0];
        for(int i=1;i<nums.length-1;i++){
            gasoline--;
            if(gasoline==-1){
                return false;
            }
            if(gasoline<nums[i]){
                gasoline=nums[i];
            }
        }
        if(gasoline>=1){
            return true;
        }
        else{
            return false;
        }
    }
}